src/MIDebugEngine/Natvis.Impl/Natvis.cs

```diff
@@ -590,6 +590,10 @@ private IVariableInformation[] ExpandVisualized(IVariableInformation variable)
                             string expressionValue = GetExpressionValue(item.Rank, variable, visualizer.ScopedNames);
                             rank = Int32.Parse(expressionValue, CultureInfo.InvariantCulture);
                         }
+                        if (rank <= 0)
+                        {
+                            throw new Exception("Invalid rank value");
```

Comment:
Will this wind up being shown to the user? If so, you need a localizable string. If not, I feel like you want to raise the error in a way that will.

Reply:
Not ideal, but I guess that sounds okay. Andrew's natvis logging stuff should hopefully address that.